### PR TITLE
feat: plugins activity log pt. 2

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Problem

Follow up to #10558 

## Changes

- Adds various activity logs for PluginConfig
- Extracts `return_activity_page` to be a reusable function rather than a static method on every viewset
- Adds `get_all_activity` function, allowing fetching multiple models at once (useful for Plugin + PluginConfig)
- Extracts `SentenceList` to its own component
- Adds `dict_changes_between` to compare across dictionaries and generate a list of changes

### What it looks like

Screenshot 2022-06-30 at 17 51 44

### Not addressed here

Plugins activity log will still need a few more parts to be even more useful, but I decided this is a good place to cut things off. What's missing:

- More activity logs e.g. attachment_updated
- Showing logs for a specific plugin (this unfortunately requires a bit of a structural refactor of the plugins frontend, hence not tackled here as was the original idea)

## How did you test this code?

Tests

### CODE REVIEW

File: posthog/api/person.py [hunk 1]
```diff
@@ -572,23 +565,7 @@ def activity(self, request: request.Request, **kwargs):
...
-                if activity_page.has_next
-                else None,
-                "previous": format_query_params_absolute_url(request, page - 1, limit, offset_alias="page")
-                if activity_page.has_previous
-                else None,
-                "total_count": activity_page.total_count,
-            },
-            status=status.HTTP_200_OK,
-        )
+        return return_activity_page(activity_page, limit, page, request)
```
Comment (pauldambra) [inline posthog/api/person.py:568]: the `self._` had hidden the slightly odd looking `return return` 

wonder if this is better as `return activity_page_response`

File: frontend/src/lib/components/ActivityLog/SentenceList.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+
+interface SentenceListProps {
+    listParts: (string | JSX.Element | null)[]
+    prefix?: string | JSX.Element | null
+    suffix?: string | JSX.Element | null
+}
+
+export function SentenceList({ listParts, prefix = null, suffix = null }: SentenceListProps): JSX.Element {
```
Comment (pauldambra) [inline frontend/src/lib/components/ActivityLog/SentenceList.tsx:9]: extracting this was well overdue 🙌  Definitely needs at least a story book page though

**Review by pauldambra — COMMENTED** [review]
didn't run it but that looks really solid 💪

**Comment by yakkomajuri** [comment]
this is getting out of hand with all the refactors. I'll put this behind a flag and get it in, and then the main thing to fix is tab URLs on the plugins/apps page.

File: frontend/src/lib/components/ActivityLog/activityLogLogic.tsx [hunk 3]
```diff
@@ -90,7 +90,7 @@ export const activityLogLogic = kea<activityLogLogicType>({
                 (pageScope === ActivityScope.PERSON && hashParams['activeTab'] === 'history') ||
                 (pageScope === ActivityScope.FEATURE_FLAG && searchParams['tab'] === 'history') ||
                 (pageScope === ActivityScope.INSIGHT && searchParams['tab'] === 'history') ||
-                (pageScope === ActivityScope.PLUGIN && searchParams['tab'] === 'history')
+                pageScope === ActivityScope.PLUGIN
```
Comment (yakkomajuri) [inline frontend/src/lib/components/ActivityLog/activityLogLogic.tsx:93]: note that this is hacky and the main thing that needs fixing in a follow up PR.

It's the reason I ended up putting this behind a flag, but not fixing this here as the PR is already bloated with refactors and this requires another one

**Comment by yakkomajuri** [comment]
Test failure is unrelated - merging.

See this other PR where it's also failing: https://github.com/PostHog/posthog/runs/7150556645?check_suite_focus=true

**Pull request merged**